src/qml/filters/gpsgraphic/ui.qml

```diff
@@ -0,0 +1,1448 @@
+                Layout.alignment: Qt.AlignRight
+            }
+            TextField {
+                id: offset_secs
+                text: '0'
+                horizontalAlignment: TextInput.AlignRight
+                validator: IntValidator {bottom: 0; top: 59;}
+                implicitWidth: 20
+                MouseArea {
+                    anchors.fill: parent
+                    onWheel: wheel_offset( wheel.angleDelta.y>0 ? 1 : -1 )
+                    onClicked: { offset_secs.forceActiveFocus() }
+                }
+                onFocusChanged: if (focus) selectAll()
+                onEditingFinished: recompute_time_offset()
+                Shotcut.HoverTip { text: qsTr('Number of seconds to add/subtract to video time to sync them.\nTip: you can use mousewheel to change values.') }
+            }
+
+            //buttons:
+            Shotcut.Button {
+                icon.name: 'media-skip-backward'
+                Shotcut.HoverTip { text: qsTr('Sync start of GPS to start of video file.\nTip: use this if you started GPS and video recording at the same time.') }
+                implicitWidth: 20
+                implicitHeight: 20
+                onClicked: { set_sec_offset_to_textfields(filter.get('auto_gps_offset_start')) }
+            }
+            Shotcut.Button {
+                icon.name: 'document-open-recent'
+                Shotcut.HoverTip { text: qsTr('Remove timezone (%1 seconds) time from video file (convert to UTC).\nTip: use this if your video camera doesn\'t have timezone settings as it will set local time as UTC.'.arg(js_tz_offset) ) }
+                implicitWidth: 20
+                implicitHeight: 20
+                onClicked: { set_sec_offset_to_textfields( parseInt(Number(filter.get('time_offset'))) + parseInt(Number(js_tz_offset)) ) }
+            }
+            Shotcut.Button {
+                icon.name: 'format-indent-less'
+                Shotcut.HoverTip { text: qsTr('Fix video start time: if file time is actually end time, press this button to subtract file length (%1 seconds) from GPS offset.'.arg(parseInt(producer.length / profile.fps)) ) }
+                implicitWidth: 20
+                implicitHeight: 20
+                onClicked: { set_sec_offset_to_textfields( parseInt(Number(filter.get('time_offset')) - producer.length / profile.fps) ); }
+            }
+            Shotcut.Button {
+                icon.name: 'media-playback-pause'
+                Shotcut.HoverTip { text: qsTr('Sync start of GPS to current video time.\nTip: use this if you recorded the moment of the first GPS fix.') }
+                implicitWidth: 20
+                implicitHeight: 20
+                onClicked: { set_sec_offset_to_textfields(filter.get('auto_gps_offset_now')) }
+            }
+            Shotcut.Button {
+                icon.name: 'document-save'
+                Shotcut.HoverTip { text: qsTr('Save the current offset as default to use in other GPS filters.') }
```

Comment:
This does not match how we have settings in other filters/fields. In other (similar) situations, we just save the setting every time the field is edited. Then, we automatically restore it when the field is populated for the first time. Do you think that would work for you here? Essentially, we could just remove this button and always save the value for the user when they edit the field.

Reply:
Functionally that could work, but I think users would just be confused and pressing reset would set it to 0 or to the saved state? 

This is how I use it (or at least why I think it's useful): have 2-3 filter instances: one with map, one with altitude, one with speed. You vaguely sync one of them and copy that value to the others. After a few minutes you notice the speed doesn't really sync with a stop you make so you tweak a few more second: now you just click the save button then the restore one for the other 2 instances (in reality I actually ended with 5 graphs and 1 text a few days ago so really used this).

Reply:
Your use case makes sense. And I think we should support it. But I think we should try to find another way. The "save" icon opens a file dialog in other places in Shotcut. So this behavior might surprise some people.

Have you see the ClockSpinner control used in the Timer filter? I think we could adapt that to add days:
https://github.com/mltframework/shotcut/blob/master/src/qml/filters/timer/ClockSpinner.qml
Then, we could modify the control to support copy/paste. 

In order to support your use case, the user could select the ClockSpinner from one filter, hit CTL-C, then switch to another filter and hit CTL-V.

This proposal would have the benefit of making the line more compact. 

How would you feel about trying to implement this? 

I hope you are not discouraged by my comments or feel like they are nit-picks. I am interested in this filter and just want to help make it as good as it can be.

Reply:
I'm not a fan of the ClockSpinner control, I find it very hard to change/increment anything specific inside (to add 1 min to it it requires just too many steps/clicks/typing). 
I find the mouse wheel up/down for each division extremely important and I really used it every time I synced the GPS files so far. Yes, the control would solve the need for the save/restore and make the line more compact, but I'd rather remove the save functionality and the days division than the individual minutes/seconds boxes. I think the latest update with removing of the "days", "m:" etc texts and replacing them with ":" makes the line compact enough to not worry about the length anymore.

Reply:
> I'm not a fan of the ClockSpinner control,

Ok. It was worth a try.

I still do not support the inconsistent use of the save button. How about, instead of saving it as a setting, use standard copy/paste buttons to copy/paste the values from the clipboard? That would be more intuitive than "saving" the values, I think.

Reply:
Adding copy + paste is just a bit too many buttons and looks cluttered for my taste, but I also understand the save button issue, it would be very unintuitive therefore unused.
Now that I think about it, the whole copy-paste thing is not that important. I think it's best I remove it and wait to see if users complain/have some other ideas about it. In all honesty it's barely cutting down a few seconds of manual change so not vital by any means.

Reply:
> Now that I think about it, the whole copy-paste thing is not that important. 

I can support that. After we get this merged, we can continue to make improvements.